### PR TITLE
PointerTy

### LINKED ISSUE #92: Create `glu::types::PointerTy` (Pointer type)

### PR DESCRIPTION
**Pull request by yunsecode**
Closes #92

### CODE REVIEW

File: include/AST/Types/PointerTy.hpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#ifndef GLU_AST_TYPES_POINTERTY_HPP
+#define GLU_AST_TYPES_POINTERTY_HPP
+
+#include "TypeBase.hpp"
+
+namespace glu::types {
+
+/// @brief Pointer is a class that represents the pointer type in the AST.
+class PointerTy : public TypeBase {
+public:
```
Comment (LindonAliu) [inline include/AST/Types/PointerTy.hpp:10]: maybe its intersent to have the type of the pointed value ?

File: include/AST/Types/PointerTy.hpp
Comment (LindonAliu) [inline include/AST/Types/PointerTy.hpp:1]: run clang format pls

File: include/AST/Types/PointerTy.hpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+namespace glu::types {
+
+/// @brief Pointer is a class that represents the pointer type in the AST.
+class PointerTy : public TypeBase {
+private:
+    TypeBase * const _type;
+
+public:
+    /// @brief Constructor for the Pointer class.
+    PointerTy() : TypeBase(TypeKind::PointerTyKind) { }
```
Comment (Snowy1803) [inline include/AST/Types/PointerTy.hpp:15]: set the field and take the pointee in ctor

File: include/AST/Types/PointerTy.hpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    /// @param type The type to check.
+    /// @return Returns `true` if the type is a `PointerTy`, `false` otherwise.
+    static bool classof(TypeBase const *type)
+    {
+        return type->getKind() == TypeKind::PointerTyKind;
+    }
+
+    /// @brief Getter for the type this pointer points to.
+    /// @return Returns the type this pointer points to.
+    TypeBase *getType() const { return _type; }
```
Comment (Snowy1803) [inline include/AST/Types/PointerTy.hpp:30]: getPointee is a better name (you can rename the field as well). this is the pointee type, type.getType() is confusing

**Review by Snowy1803 — APPROVED** [review]

**Pull request merged**